lib/fennec/turn.ex

```diff
@@ -2,13 +2,19 @@ defmodule Fennec.TURN do
   @moduledoc false
   # This module defines a struct used as TURN protocol state.
 
-  defstruct allocation: nil, permissions: %{}, channels: [],
-            nonce: nil, realm: nil
+  alias Jerboa.Format
+  alias Fennec.TURN.Channel
+
+  defstruct allocation: nil, permissions: %{}, channels: {%{}, %{}},
+    nonce: nil, realm: nil
+
+  @permission_lifetime 5 * 60 # MUST be 5mins
 
   @type t :: %__MODULE__{
     allocation: nil | Fennec.TURN.Allocation.t,
     permissions: %{peer_addr :: Fennec.ip => expiration_time :: integer},
-    channels: [],
+    channels: {%{peer :: Fennec.address => Channel.t},
+               %{Format.channel_number  => Channel.t}},
```

Comment:
I was gonna say that were not likely to have many channels 😃  The most intensive a use case I can imagine is like a multiplayer game where it's not gonna see more than like 16 gamers at a time or something. I imagine. I don't know.